TypeCobol/Compiler/Scanner/MultilineScanStateSnapshot.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TypeCobol.Compiler.Scanner
+{
+    /// <summary>
+    /// Snapshot of a  MultilineScanState
+    /// </summary>
+    public class MultilineScanStateSnapshot : MultilineScanState
+    {
+        public MultilineScanStateSnapshot(MultilineScanState currentState) : base(currentState)
+        {
+            _AfterExecSql = currentState.AfterExecSql;
+            _AfterExec = currentState.AfterExec;
+            _AfterExecTranslatorName = currentState.AfterExecTranslatorName;
+            _AfterExecStatementText = currentState.AfterExecStatementText;
+            _AfterPicture = currentState.AfterPicture;
+            _AfterCommentEntryKeyword = currentState.AfterCommentEntryKeyword;
+            _AfterCommentEntryKeywordPeriod = currentState.AfterCommentEntryKeywordPeriod;
+            _AfterCommentEntry = currentState.AfterCommentEntry;
+            _AfterFUNCTION = currentState.AfterFUNCTION;
+            _AtBeginningOfSentence = currentState.AtBeginningOfSentence;
+        }
+
+        private bool _AfterExecSql;
+        public override bool AfterExecSql
+        {
+            get
+            {
+                return AfterExecSql;
```

Comment:
Typo, beware the property is now recursive !

Reply:
Yep, yes